### DISCUSSION TITLE
After deleting ES indexes and restarting the OAP service and UI, some indexes cannot be rebuilt normally.

### DISCUSSION THREAD

**Discussion by anjoy8** [apache/skywalking#8271] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

**First of all**, I started the oap and ui services with the docker image of 8.9.0.
 
**Then**, an ES index with today(20211209) as suffix will be generated.

**Next**, I delete these indexes and restart OAP service.

**Finally**, some indexes cannot be created, such as sw_service_traffic.

eg.

> Caused by: java.lang.RuntimeException: {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"sw_network_address_alias","index_uuid":"_na_","index":"sw_network_address_alias"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"sw_network_address_alias","index_uuid":"_na_","index":"sw_network_address_alias"},"status":404}

### What you expected to happen

I hope the project can start normally and create the ES index of index.

### How to reproduce

**First of all**, I started the oap and ui services with the docker image of 8.9.0.
```
docker run --name skywalking-oap -p 11800:11800 -p 12800:12800 -d    -e SW_STORAGE=elasticsearch -e SW_STORAGE_ES_CLUSTER_NODES=elasticsearch:9200   apache/skywalking-oap-server:8.9.0
 
docker run --name skywalking-ui --restart always -p 8080:8080  -d -e SW_OAP_ADDRESS=http://elasticsearch:12800 apache/skywalking-ui:8.9.0
```
**Then**, an ES index with today(20211209) as suffix will be generated.
**Next**, I delete these indexes and restart OAP service.
**Finally**, some indexes cannot be created, such as sw_service_traffic.

![image](https://user-images.githubusercontent.com/28941816/145332332-fb235cfd-bfc8-4aaa-94ae-08f282bb50fe.png)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
What you did on your private env is mostly on you. The community can't tell what is status there. Check the logs, read this carefully.

**Comment by anjoy8** [comment]
By debugging the source code, I found that when I deleted the index but did not delete the template, restarting the oap service would not create the corresponding index.
The relevant code is here:

``` 
// StorageEsInstaller
boolean shouldUpdateTemplate = !esClient.isExistsTemplate(tableName);
if (shouldUpdateTemplate) {
  // ...
}
```

**Reply by anjoy8** [reply]
I want to know, why is it designed like this?

**Reply by wu-sheng** [reply]
Because we can't tell why you removed or partial removing something.
The codes are for clean env, rather than your randomly changing.

**Reply by anjoy8** [reply]
Sorry, if there are too many indexes one day, it needs to be deleted.

**Reply by wu-sheng** [reply]
Don't know what do you mean. Once you deleting on your own, it is not community responsibility to take care.
Apache 2.0 License have granted you to do thing privately.

**Reply by anjoy8** [reply]
Sorry, maybe I didn't describe it clearly. It has nothing to do with code and community.
I just want to say that template information needs to be deleted after deleting the index.

**Reply by BritneyStars** [reply]
I have ame issue . I debuged and found  the  shouldUpdateTemplate  is always false anywhere the template exists. the oap server version is 8.9.1. But when I use oap server 8.5.0 debug , the shouldUpdateTemplate  have true and false ,and the shouldUpdateTemplate  value  is consistent with the actual situation of elasticsearch template

**Reply by CCJason-web** [reply]
> 

I met the same issue like yours. I also debug to the function of createTimeSeriesTable, and I also didn't know why hardcode the shouldUpdateTemplate. So from your information that should I install 8.5.0 and then upgrate it to 8.9.1? I tried this but also failed.

**Reply by CCJason-web** [reply]
Event thought I downgraded the version to 8.3.0 and then upgrade to 8.9.1 still failed. So sad. Somebody can help me?

**Reply by CCJason-web** [reply]
fianlly, I delete all templates and index templates then it works.

**Reply by admlke** [reply]
How to delete all templates and indexes？I have the same problem
